Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs

```diff
@@ -0,0 +1,228 @@
+                                                   Value = Round(sample.AIDependencyCallsFailedPerSecond),
+                                                   Weight = 1
+                                               },
+                                           new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Dependency Calls Succeeded/Sec",
+                                                   Value = Round(sample.AIDependencyCallsSucceededPerSecond),
+                                                   Weight = 1
+                                               },
+                                            new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Exceptions/Sec",
+                                                   Value = Round(sample.AIExceptionsPerSecond),
+                                                   Weight = 1
+                                               }
+                                       };
+
+                metricPoints.AddRange(sample.PerfCountersLookup.Select(counter => new MetricPoint { Name = counter.Key, Value = Round(counter.Value), Weight = 1 }));
+
+                var dataPoint = new MonitoringDataPoint
+                                    {
+                                        Version = this.version,
+                                        InstrumentationKey = instrumentationKey,
+                                        Instance = this.instanceName,
+                                        Timestamp = sample.EndTimestamp,
+                                        Metrics = metricPoints.ToArray()
+                                    };
+
+                monitoringPoints.Add(dataPoint);
+            }
+
+            var serializer = new DataContractJsonSerializer(typeof(MonitoringDataPoint[]));
+            serializer.WriteObject(bodyStream, monitoringPoints.ToArray());
+
+            bodyStream.Position = 0;
+
+            return bodyStream;
+        }
+        
+        private HttpWebResponse SendRequest(string httpVerb, string path, MemoryStream body, bool enableRetry)
+        {
+            var requestUri = string.Format(CultureInfo.InvariantCulture, "{0}/{1}", this.ServiceUri.AbsoluteUri.TrimEnd('/'), path.TrimStart('/'));
+
+            int attemptMaxCount = enableRetry ? RetryCount : 1;
+            int attempt = 0;
+            while (attempt < attemptMaxCount)
+            {
+                try
+                {
+                    var request = WebRequest.Create(requestUri) as HttpWebRequest;
+                    request.Method = httpVerb;
+                    request.Timeout = (int)this.timeout.TotalMilliseconds;
+
+                    if (body != null)
+                    {
+                        body.Position = 0;
```

Comment:
Good point, but not applicable anymore, eliminated the intermediary buffer. Thanks!